File: prisma/refs/py/string-encode-decode.py
class Codec:
    def encode(self, strs):
        return "".join(f"{len(s)}#{s}" for s in strs)

    def decode(self, s):
        out, i = [], 0
        while i < len(s):
            j = s.index("#", i)
            length = int(s[i:j])
            out.append(s[j + 1:j + 1 + length])
            i = j + 1 + length
        return out

File: prisma/refs/string-encode-decode.ts
export class Codec {
  encode(strs: string[]): string {
    return strs.map((s) => `${s.length}#${s}`).join("");
  }

  decode(s: string): string[] {
    const out: string[] = [];
    let i = 0;
    while (i < s.length) {
      const hash = s.indexOf("#", i);
      const len = Number(s.slice(i, hash));
      out.push(s.slice(hash + 1, hash + 1 + len));
      i = hash + 1 + len;
    }
    return out;
  }
}
